overlord/state/notices.go

```diff
@@ -248,9 +346,23 @@ func (s *State) AddNotice(userID *uint32, noticeType NoticeType, key string, opt
 			expireAfter:   defaultNoticeExpireAfter,
 			occurrences:   1,
 		}
+		notice.setRepeatCheckValue(options.RepeatCheckData)
 		s.notices[uniqueKey] = notice
 		newOrRepeated = true
 	} else {
+		if options.RepeatCheck != nil {
+			repeatOk, newRepeatCheckData, err := options.RepeatCheck(notice, options)
+			if err != nil {
+				return "", err
+			}
+			notice.setRepeatCheckValue(newRepeatCheckData)
+			// drop notice
+			if !repeatOk {
+				return notice.id, nil
```

Comment:
shouldn't the flow be the same as when the check below `options.RepeatAfter == 0 || now.After(notice.lastRepeated.Add(options.RepeatAfter)` is false, instead of exiting here?

Reply:
Updated, Thanks!

Reply:
We might end up with a slightly confusing `notice.repeatAfter` value though. If we pass a `RepeatAfter` value that should normally repeat (which is always attached to the notice.repeatCheck) but its behavior was overridden by `RepeatCheck`.

Reply:
my initial thought was actually to check RepeatAfter if set before RepeatCheck,  which woudl be slightly more consistent? but indeed if somebody sets RepeatCheck the point is to be more flexible that just RepeatAfter, but there is always the option not to set the latter

Reply:
Does it make sense then to also make `RepeatAfter` and `RepeatCheck` mutually exclusive? like what is done for `RepeatCheckData` and `RepeatCheck` [here](https://github.com/snapcore/snapd/pull/13576/files#diff-4fc1b33549a08f2c1e8c5ae6ebdd8a0b859cc5e93b7208bc6656e02c208a9183R389-R391)?

Reply:
no, I'm also not sure that the check you pointed to make sense either. Like for our use case you might want to always pass the current change status a RepeatCheckData but it gets ignored if we are not repeating it.